hyperspy/_signals/eels.py

```diff
@@ -1220,10 +1219,10 @@ def kramers_kronig_analysis(self,
             # Kramers Kronig Transform:
             # We calculate KKT(Im(-1/epsilon))=1+Re(1/epsilon) with FFT
             # Follows: D W Johnson 1975 J. Phys. A: Math. Gen. 8 490
-            # Use a size that is a power of two to speed up the fft and
+            # Use an optimal FFT size to speed up the calculation, and
             # make it double the closest upper value to workaround the
             # wrap-around problem.
-            esize = 2 * closest_power_of_two(axis.size)
+            esize = 2 * optimal_fft_size(axis.size)
```

Comment:
Maybe this should be `optimal_fft_size(2 * axis.size)`

Reply:
I guess `optimal_fft_size(2 * axis.size)` should more sense and give slightly smaller number. I think you reverted a commit here when trying to debug the samfire, so I am not sure if you want to reintroduce this change or not.